### PR TITLE
LF: Move transaction validation is in is own object.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
+ cleanup
 + rewrite value comparison. Do not use Equality as it may not be symetric.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Validation.scala [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+                } &&
+                loop((values(recordedFields) zip values(recordedFields)) ++: rest)
+            case (
+                  ValueVariant(recordedTyCon, recordedVariant, recordedValue),
+                  ValueVariant(replayedTyCon, replayedVariant, replayedValue),
+                ) =>
+              recordedTyCon == replayedTyCon &&
+                recordedVariant == replayedVariant &&
+                loop((recordedValue, replayedValue) +: rest)
+            case (ValueList(recordedValues), ValueList(replayedValues)) =>
```
Comment (cocreature) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Validation.scala:89]: I know this didn’t change but I don’t understand why we don’t have to check the length here.

**Review by cocreature — APPROVED** [review]
lgtm assuming this just shuffling things around

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Validation.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Validation.scala:89]: It is in fact changed (I update the PR description).
We have to check the length (you already catch that in another PR that I try to slice apart)
Thanks.

**Review by cocreature — APPROVED** [review]
Thanks! Took another look given that there have been changes. My only worry is still the lack of testing but you’re aware of that. I think it’s probably a good idea to prioritize that once the type erasure changes have landed.

**Pull request merged**